internal/sdkprovider/service/vpc/org_vpc_peering_connection.go

```diff
@@ -0,0 +1,145 @@
+package vpc
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	avngen "github.com/aiven/go-client-codegen"
+	"github.com/aiven/go-client-codegen/handler/organizationvpc"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/aiven/terraform-provider-aiven/internal/common"
+)
+
+var (
+	pollInterval = common.DefaultStateChangeMinTimeout
+	pollDelay    = 1 * time.Second
+)
+
+func createPeeringConnection(
+	ctx context.Context,
+	orgID, orgVpcID string,
+	client avngen.Client,
+	d *schema.ResourceData,
+	req organizationvpc.OrganizationVpcPeeringConnectionCreateIn,
+) (*organizationvpc.OrganizationVpcGetPeeringConnectionOut, error) {
+	pc, err := client.OrganizationVpcPeeringConnectionCreate(
+		ctx,
+		orgID,
+		orgVpcID,
+		&req,
+	)
+	if err != nil || pc.PeeringConnectionId == nil {
+		return nil, fmt.Errorf("error creating VPC peering connection: %w", err)
+	}
```

Comment:
Nice catch with `nil` connection id, but the error would be confusing.

Reply:
Split it into two separate checks